pkg/operator/controller/controller.go

```diff
@@ -131,14 +131,20 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 			} else if err := r.enforceEffectiveHighAvailability(ingress, infraConfig); err != nil {
 				errs = append(errs, fmt.Errorf("failed to enforce the effective HA configuration for clusteringress %s: %v", ingress.Name, err))
 			} else if ingress.DeletionTimestamp != nil {
-				// Handle deletion.
+				// Handle ingress deletion and finalization.
 				if err := r.ensureIngressDeleted(ingress, dnsConfig, infraConfig); err != nil {
 					errs = append(errs, fmt.Errorf("failed to ensure ingress deletion: %v", err))
 				}
+				if err := r.finalizeClusterIngress(ingress); err != nil {
```

Comment:
If `ensureIngressDeleted` returns an error, `finalizeClusterIngress` is still called. Shouldn't the ingress finalizer only be removed once we're certain we successfully cleaned up?

Reply:
ack